build.ps1

```diff
@@ -3,9 +3,7 @@ Param(
     [Parameter(Position=1)]
     [String] $Target = "build",
     [String] $Build = '',
-    [String] $ParentImageVersion = '3131.vf2b_b_798b_ce99-4',
-    [String] $BuildNumber = '1',

```

Comment:
This is required: the build number (e.g. the tag for jenkins/inbound-agent) can differ from the parent's image build number (e.g. the tag of the jenkins/agent used as a parent image).

We try to keep them synchronized but it can be different technically speaking

Reply:
There wasn't any build number in the make.ps1 previously: https://github.com/jenkinsci/docker-inbound-agent/blob/c58cdac1fdc0d3f339e94ec63220847e60c00acf/make.ps1

Reply:
I think having a mechanism to be able to get multiple builds for each release would be nice to have but it wasn't there previously. I propose opening an issue about it, WDYT @dduportal?

Reply:
https://github.com/jenkinsci/docker-inbound-agent/blob/c58cdac1fdc0d3f339e94ec63220847e60c00acf/make.ps1#L6C15-L7

There was 2 "versions" provided: one for the parent and one for the "current tag".

#363 broke this behavior and I was assuming the fixup here was correcting this behavior.

Reply:
> I think having a mechanism to be able to get multiple builds for each release would be nice to have but it wasn't there previously. I propose opening a distinct issue about it, WDYT @dduportal?

That is not what my comment is about :) My comment is about the ability to build and deploy the proper version based on the tag which created the deployment build on trusted.ci, tag that can be different than the parent image tag value

Reply:
Tags pushed from a build before the refactoring (with docker-agent version 3131.vf2b_b_798b_ce99**-3**), https://trusted.ci.jenkins.io:1443/job/Containers/job/Agents/job/docker-inbound-agent/job/3131.vf2b_b_798b_ce99-3/2/:
```
3131.vf2b_b_798b_ce99-3-jdk17-nanoserver-1809
jdk17-windowsservercore-ltsc2019

3131.vf2b_b_798b_ce99-3-jdk17-windowsservercore-ltsc2019
jdk11-nanoserver-1809

3131.vf2b_b_798b_ce99-3-jdk11-nanoserver-1809
nanoserver-1809
3131.vf2b_b_798b_ce99-3-nanoserver-1809
```

Wrong output from the current master branch:
```
= PREPARE: List of images and tags to be processed:
{
"jdk11-windowsservercore-ltsc2019":  {
    "Tags":  [
                "jdk11-windowsservercore-ltsc2019",
                "3131.vf2b_b_798b_ce99-4-1-jdk11-windowsservercore-ltsc2019",
                "windowsservercore-ltsc2019"
            ]
    },
"jdk17-nanoserver-1809":  {
    "Tags":  [
                "jdk17-nanoserver-1809",
                "3131.vf2b_b_798b_ce99-4-1-jdk17-nanoserver-1809"
            ]
    },
"jdk17-windowsservercore-ltsc2019":  {
    "Tags":  [
                "jdk17-windowsservercore-ltsc2019",
                "3131.vf2b_b_798b_ce99-4-1-jdk17-windowsservercore-ltsc2019"
            ]
    },
"jdk11-nanoserver-1809":  {
    "Tags":  [
                "jdk11-nanoserver-1809",
                "3131.vf2b_b_798b_ce99-4-1-jdk11-nanoserver-1809",
                "nanoserver-1809"
            ]
    }
}
```

Correct output of https://ci.jenkins.io/job/Packaging/job/docker-inbound-agent/job/PR-374/2/:

```
= PREPARE: List of images and tags to be processed:
{
"jdk11-windowsservercore-ltsc2019":  {
    "Tags":  [
                "jdk11-windowsservercore-ltsc2019",
                "3131.vf2b_b_798b_ce99-4-jdk11-windowsservercore-ltsc2019",
                "windowsservercore-ltsc2019"
            ]
    },
"jdk17-nanoserver-1809":  {
    "Tags":  [
                "jdk17-nanoserver-1809",
                "3131.vf2b_b_798b_ce99-4-jdk17-nanoserver-1809"
            ]
    },
"jdk17-windowsservercore-ltsc2019":  {
    "Tags":  [
                "jdk17-windowsservercore-ltsc2019",
                "3131.vf2b_b_798b_ce99-4-jdk17-windowsservercore-ltsc2019"
            ]
    },
"jdk11-nanoserver-1809":  {
    "Tags":  [
                "jdk11-nanoserver-1809",
                "3131.vf2b_b_798b_ce99-4-jdk11-nanoserver-1809",
                "nanoserver-1809"
            ]
    }
}
```

Reply:
> https://github.com/jenkinsci/docker-inbound-agent/blob/c58cdac1fdc0d3f339e94ec63220847e60c00acf/make.ps1#L6C15-L7
> 
> There was 2 "versions" provided: one for the parent and one for the "current tag".
> 
> #363 broke this behavior and I was assuming the fixup here was correcting this behavior.

Just saw this comment, will reintroduce the build number.

Reply:
> > I think having a mechanism to be able to get multiple builds for each release would be nice to have but it wasn't there previously. I propose opening a distinct issue about it, WDYT @dduportal?
> 
> That is not what my comment is about :) My comment is about the ability to build and deploy the proper version based on the tag which created the deployment build on trusted.ci, tag that can be different than the parent image tag value

Reading again the code of make.ps1 from before the refactoring, and the corresponding logs on trusted.ci.jenkins.io I fail to see where this build number could be. Do you have an example to show?

Reply:
I guess we both missed there was a split of the "parent image" tag which goal was to get rid of the build number from "parent" (and then the new one was added).

Damn, that would be way simpler once both repo will be merged 😅

Sorry, my explanations were a bit short but I think you got it 👍

Reply:
> I guess we both missed there was a split of the "parent image" tag which goal was to get rid of the build number from "parent" (and then the new one was added).

I'm not sure there was one before:
https://github.com/jenkinsci/docker-inbound-agent/blob/c58cdac1fdc0d3f339e94ec63220847e60c00acf/Jenkinsfile#L58

https://github.com/jenkinsci/docker-inbound-agent/blob/c58cdac1fdc0d3f339e94ec63220847e60c00acf/make.ps1#L189-L228

Reply:
Example from [one of my previous comment](https://github.com/jenkinsci/docker-inbound-agent/pull/374#discussion_r1278597869), this build https://trusted.ci.jenkins.io:1443/job/Containers/job/Agents/job/docker-inbound-agent/job/3131.vf2b_b_798b_ce99-3/2/ is the second one (where is the first?) but the generated tags have `3` in them (docker-agent parent image build number)

Reply:
Seen together interactively, it should be fixed as we agreed on the problem statement and a solution. Thanks!